core/lib/src/data/bytes.rs

```diff
@@ -0,0 +1,88 @@
+use std::borrow::Cow;
+use std::io;
+
+use crate::data::{Capped, FromData, Limits, Outcome, N};
+use crate::form::{DataField, Errors, FromFormField, ValueField};
+use crate::http::{ContentType, Status};
+use crate::{fs::FileName, outcome::IntoOutcome, Data, Request};
+
+#[derive(Debug)]
+pub struct Bytes<'v> {
+    pub file_name: Option<&'v FileName>,
+    pub content_type: Option<ContentType>,
+    pub content: Cow<'v, [u8]>,
+}
+
+impl<'v> Bytes<'v> {
+    async fn from<'a>(
+        req: &Request<'_>,
+        data: Data<'_>,
+        file_name: Option<&'a FileName>,
+        content_type: Option<ContentType>,
+    ) -> io::Result<Capped<Bytes<'a>>> {
+        let limit = { content_type.as_ref() }
+            .and_then(|ct| ct.extension()?.limits().find(&["file", ext.as_str()]))
```

Comment:
I think there's a mistake here, shouldn't be
```rs
            .and_then(|ct| ct.extension())
            .and_then(|ext| req.limits().find(&["file", ext.as_str()]))
```
?

Reply:
Sorry for the delay, been busy working.
There is no need for a second `.and_then(...)` because the `?` after `.extension()` extracts the value from `Option::Some(T)` or returns `Option::None` depending on which variant it happened to be, allowing for more concise code.
Apparently it was [stabilized back in 2016](https://github.com/rust-lang/rust/pull/36995), feels newer. There's a [section in the book about it](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator).